src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -580,12 +762,12 @@
 (def user-query->logical-obj
   "Keypairs of the stringified subquery keyword (found in user defined queries) to the
   appropriate plan node"
-  {"select_nodes" (assoc nodes-query :subquery? true)
-   "select_resources" (assoc resources-query :subquery? true)
-   "select_params" (assoc resource-params-query :subquery? true)
-   "select_facts" (assoc facts-query :subquery? true)
+  {"select_facts" (assoc facts-query :subquery? true)
+   "select_fact_contents" (assoc fact-contents-query :subquery? true)
+   "select_nodes" (assoc nodes-query :subquery? true)
    "select-latest-report" (assoc latest-report-query :subquery? true)
```

Comment:
Nope, its used to achieve the 'latest_report?' field which is how its tested as well, its a little weird I agree but its how it is today. At some point we should remove the specialisation.